src/client/proto.rs

```diff
@@ -196,22 +196,168 @@ where
     }
 }
 
-#[derive(Debug, Deserialize, Default)]
-pub struct Pcrs<'a> {
+#[derive(Debug, Copy, Clone)]
+#[repr(u32)]
+pub enum PcrAlgo {
+    Sha1,
+    Sha256,
+    Sha384,
+    Sha512,
+    Unknown(u32),
+}
+
+impl From<u32> for PcrAlgo {
+    fn from(x: u32) -> Self {
+        // FIXME: due to https://github.com/rust-lang/rust/issues/60553 we map
+        // identity map raw values into corresponding enum constants. Rust if free
+        // to assign any discrimant to each variant.
+        match x {
+            0x04 => Self::Sha1,
+            0x0b => Self::Sha256,
+            0x0c => Self::Sha384,
+            0x0d => Self::Sha512,
+            _ => Self::Unknown(x),
+        }
+    }
+}
+
+impl From<PcrAlgo> for u32 {
+    fn from(x: PcrAlgo) -> Self {
+        match x {
+            PcrAlgo::Sha1 => 0x04,
+            PcrAlgo::Sha256 => 0x0b,
+            PcrAlgo::Sha384 => 0x0c,
+            PcrAlgo::Sha512 => 0x0d,
+            PcrAlgo::Unknown(x) => x,
+        }
+    }
+}
+
+impl<'de> serde::Deserialize<'de> for PcrAlgo {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::de::Deserializer<'de>,
+    {
+        struct Visitor;
+        impl<'de> serde::de::Visitor<'de> for Visitor {
+            type Value = PcrAlgo;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("an integer")
+            }
+
+            fn visit_u32<E>(self, v: u32) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                Ok(PcrAlgo::from(v))
+            }
+        }
+
+        deserializer.deserialize_u32(Visitor)
+    }
+}
+
+impl serde::Serialize for PcrAlgo {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_u32((*self).into())
+    }
+}
+
+impl fmt::Display for PcrAlgo {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Sha1 => write!(f, "sha1"),
+            Self::Sha256 => write!(f, "sha256"),
+            Self::Sha384 => write!(f, "sha384"),
+            Self::Sha512 => write!(f, "sha512"),
+            Self::Unknown(u) => write!(f, "unknown (0x{:x})", u),
```

Comment:
```suggestion
            Self::Unknown(u) => write!(f, "unknown (0x{:04x})", u),
```
or `"unknown ({:#06x})"`. Full length u16 should be more readable than shortened version.